src/www/firewall_rules_edit.php

```diff
@@ -74,9 +75,11 @@ function is_posnumericint($arg) {
     return (is_numericint($arg) && $arg[0] != '0' && $arg > 0);
 }
 
-
 $a_filter = &config_read_array('filter', 'rule');
 
+// Whether POST or GET, the URL should still have a valid $_GET['id'] field controlling the interface edited
+// It's only validated on POST, but that's safe because only the initial page load is a GET and config only updates upon POST
+$interface_from_url = (empty($_GET['if']) || !is_string($_GET['if'])) ? "" : $_GET['if'];
```

Comment:
this isn't validation as it's assigning any value... benefit vs. cost... and is likely validated below correctly?

Reply:
It is validating a value - in an important sense. The value shown to the user, and the interface they understand they are editing, is based on this.  As a general principle it shouldn't be possible to have contradictory values this way. Or put another way, if `&if=` is how the interface being edited is specified, and the POST comes back with $_GET containing `&if=lan` but `$_POST['interface']=wan`, then something's wrong and we shouldn't process the form. 

Also noted that the menu title, **which the user is shown** to confirm the interface being edited, is generated from the URI via $_GET (head.inc/19: `$menu->getItems($_SERVER['REQUEST_URI']);`). But **the interface actually configured** is derived from $_POST. So if the two differ "somehow", the interface presented as being edited and the interface actually edited will silently differ. Very not good.

Cost of checking they;'re consistent? Seems tiny by any standard. Assign it to a variable at the start, use that variable as the authoritative source of "interface being edited" throughout form generation, and check $_POST['interface'] is still consistent with it on form submission. That's all. Trivial to check.

For those reasons I've added this check. It ensures that at least, the interface to be edited, and the interface the user is told will be modified, are the same. Which at best shouldn't hurt and is no bad thing to add if cautious. 

Hopefully you can agree - or at least decide it's not objectionable to check anyhow?

Reply:
It looks like this tries to fix something in the middle. We have five cases:

1. if= not set, means floating
2. if= set, doesn't load page properly
3. if=validinterface set, means floating
4. if=invalidinterface set, doesn't load page properly
5. if=FloatingRules set, means floating

so we "fix" 2 but not 4 ? why? what is the strategy to deal with invalid input other than assuming "floating" ?

Reply:
- Case 1 shouldn't ever be possible. If floating, the uri will be `&if=FloatingRules` and the same URI will be loaded after POST. It's never missing if things are working correctly.
- Case 2 also incorrect. It loads the page fully (albeit without a breadcrumb but total omission => not obvious). And the page fully works. It will allow a page to be edited and saved, and save it against whatever interface is in $_POST even if &if= is empty.
- Case 4 also incorrect., behaves same as case 2.

This PR fixes all of them, by ensuring &if is valid, the one or multiple interfaces in $_POST['interface'] are valid, and that &if is either FloatingRules or the single (valid) interface in $_POST['interface'].

Reply:
Thanks for explaining what I said. Now that we are on the same page I don't agree with the proposed fix for 1, 2 and 4.

Reply:
To be more precise: there's no use validating only POST if we go so far as to present a faulty form to the user on a faulty GET request, because the accepted GET request is the root of the problem.